app/controllers/actions/IdentifierAction.scala

```diff
@@ -37,41 +41,49 @@ class AuthenticatedIdentifierAction @Inject() (
   val parser: BodyParsers.Default
 )(implicit val executionContext: ExecutionContext)
     extends IdentifierAction
-    with AuthorisedFunctions {
+    with AuthorisedFunctions
+    with Logging {
 
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId =>
-          block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
-    } recover {
+    authorised(AuthProviders(GovernmentGateway) and ConfidenceLevel.L50)
+      .retrieve(Retrievals.internalId and Retrievals.allEnrolments and Retrievals.affinityGroup) {
+        case Some(internalId) ~ enrolments ~ Some(affinity) => getSubscriptionId(request, enrolments, internalId, affinity, block)
+        case _ =>
+          logger.warn("Unable to retrieve internal id or affinity group")
+          throw AuthorisationException.fromString("Unable to retrieve internal id or affinity group")
+      } recover {
       case _: NoActiveSession =>
         Redirect(config.loginUrl, Map("continue" -> Seq(config.loginContinueUrl)))
       case _: AuthorisationException =>
         Redirect(routes.UnauthorisedController.onPageLoad)
     }
   }
 
-}
-
-class SessionIdentifierAction @Inject() (
-  val parser: BodyParsers.Default
-)(implicit val executionContext: ExecutionContext)
-    extends IdentifierAction {
+  private def getSubscriptionId[A](request: Request[A],
+                                   enrolments: Enrolments,
+                                   internalId: String,
+                                   affinityGroup: AffinityGroup,
+                                   block: IdentifierRequest[A] => Future[Result]
+  ): Future[Result] = {
 
-  override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
+    val fatcaEnrolment  = config.enrolmentKey
```

Comment:
Looks like these two variables are only used in the for comprehension below so I don't think they need to be variables and you could just call the config/object directly in the comprehension? e.g.:

```
  enrolment <- enrolments.getEnrolment(config.enrolmentKey)
  id <- enrolment.getIdentifier(IdentifierType.FATCAID)
```